src/legacy/core_plugins/metrics/server/lib/vis_data/helpers/unit_to_seconds.test.js

```diff
@@ -0,0 +1,120 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import getUnitValue, { parseInterval, convertIntervalToUnit, getSuitableUnit } from './unit_to_seconds';
+
+describe('unit_to_seconds', () => {
+  describe('parseInterval()', () => {
+    test('should parse interval (positive tests)', () => {
+      expect(parseInterval('1m')).toEqual({
+        value: 1,
+        unit: 'm',
+      });
+
+      expect(parseInterval('134d')).toEqual({
+        value: 134,
+        unit: 'd',
+      });
+
+      expect(parseInterval('30M')).toEqual({
+        value: 30,
+        unit: 'M',
+      });
+    });
+
+    test('should parse interval (negative tests)', () => {
+      expect(parseInterval('gm')).toEqual({
+        value: undefined,
+        unit: undefined,
+      });
+
+      expect(parseInterval('-1d')).toEqual({
+        value: undefined,
+        unit: undefined,
+      });
+
+      expect(parseInterval('M')).toEqual({
+        value: undefined,
+        unit: undefined,
+      });
+    });
+  });
+
+  describe('convertIntervalToUnit()', () => {
+    test('should convert interval to unit (positive tests)', () => {
```

Comment:
A test should be as much specific as it allows to be.

In this case the test try to cover all positive outcomes in itself. But when something goes wrong it won't be clear what was broken: converting minutes to hours or hours to minutes or hours to seconds and so on.

The best practice is testing of individual cases separately:
`test('should convert hours to minutes', ...`
`test('should convert hours to seconds', ...`
 And so on.